src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -4058,5 +4058,38 @@ def test_vmss_create_disk_iops_mbps(self, resource_group):
                  ])
 
 
+class VMCreateAutoCreateSubnetScenarioTest(ScenarioTest):
+
+    @ResourceGroupPreparer(name_prefix='cli_test_vm_subnet')
+    def test_vm_create_auto_create_subnet(self, resource_group):
+
+        self.kwargs.update({
+            'loc': 'eastus',
+            'vm': 'vm-subnet',
+            'vnet': 'myvnet'
+        })
+
+        # Expecting no results
+        self.cmd('vm list --resource-group {rg}',
+                 checks=self.is_empty())
+        self.cmd('network vnet list --resource-group {rg}',
+                 checks=self.is_empty())
+
+        self.cmd('network vnet create --resource-group {rg} --name {vnet} --location {loc}')
+        self.cmd('vm create --resource-group {rg} --location {loc} --name {vm} --admin-username ubuntu --image UbuntuLTS --admin-password testPassword0 --authentication-type password --vnet-name {vnet}')
```

Comment:
The test didn't fail because the vnet doesn't have subnets. Assume we create a vnet with this command:

```
az network vnet create -g {} -n vnet5 --location eastasia --address-prefix 10.0.0.0/16 --subnet-name sub1 --subnet-prefix 10.0.0.0/24 
```

Then the `vm create` will fail:

```
vm create --resource-group {} --location southeastasia --name {} --admin-username ubuntu --image UbuntuLTS --admin-password testPassword0 --authentication-type password --vnet-name vnet5
Azure Error: NetcfgInvalidSubnet
Message: Subnet '{}Subnet' is not valid in virtual network 'vnet5'.
```
Because `10.0.0.0/24` is already occupied by `sub1`.